File: Data Structure/3.Linked List/Add Element at Kth Position in LL.py
# https://workat.tech/problem-solving/practice/add-kth-element-linked-list
"""
This is the ListNode class definition

class ListNode:
	def __init__(self, data=0, next=None):
		self.data = data
		self.next = next

"""


class Solution:
	def addAtkthElement(self, head: ListNode, k: int, newElement: ListNode) -> ListNode:
		# add your logic here
		t = head
		if k == 1:
			newElement.next  = head
			head = newElement
		else:
			for i in range(k-2):
				t = t.next
			
			newElement.next = t.next
			t.next = newElement
			
		while(head!=None):
			print(head.data,end=' ')
			head = head.next

File: Data Structure/3.Linked List/Add Element at Kth Position in LL.java
/** This is the ListNode class definition

class ListNode {
	int data;
	ListNode next;

	ListNode(int data) {
		this.data = data;
		this.next = null;
	}
}
**/
// https://workat.tech/problem-solving/practice/add-kth-element-linked-list
class Solution {
	ListNode addAtkthElement (ListNode head, int k, ListNode newElement) {
	    // add your logic here
		int i=1;
		ListNode t  = head;
		while(t != null){
			if(k==1){
				newElement.next = head;
				head = newElement;
				break;
			}
			if(i ==  k-1 ){
				newElement.next = t.next;
				t.next = newElement;
				break;
			}
			t = t.next;
			i+=1;
		}
		return head;
	}
}